### PR TITLE
Accept $_GET parameters &list=x,y,z (or just &list=x) and &htmlemail=1

### PR DESCRIPTION
**Pull request by lwcorp**
## Description
$_REQUEST['email'] and $_REQUEST['emailconfirm'] are already allowed, so let's also support the other two fields.

## Related Issue
https://mantis.phplist.org/view.php?id=15235

### CODE REVIEW

**Comment by michield** [comment]
:+1:

**Comment by suelaP** [comment]
@lwcorp sorry for the delay on this. I am trying to debug a problem with my local set up (not related to these changes) to be able and properly test this.

File: public_html/lists/admin/subscribelib2.php [hunk 1]
```diff
@@ -716,6 +716,10 @@ function ListAvailableLists($userid = 0, $lists_to_show = '')
     global $tables;
     if (isset($_POST['list'])) {
         $list = $_POST['list'];
+    } elseif (!isset($_POST["subscribe"]) && isset($_GET['list'])) {
+        $list_value = "signup";
+        $list_values = explode(",", $_GET["list"]);
```
Comment (xh3n1) [inline public_html/lists/admin/subscribelib2.php]: @lwcorp Looks good, but I would suggest using (int)$_GET["list"]  to make sure that the parameter is Integer.

Comment (lwcorp) [reply public_html/lists/admin/subscribelib2.php]: Thanks, but my feature allows multiple lists, thus the "explode" command.
Please enter the link in the description and you'll see the example of ?p=subscribe&list=x,y,z

Comment (xh3n1) [reply public_html/lists/admin/subscribelib2.php]: but x, y, z are expected to be Integers, right?

Comment (lwcorp) [reply public_html/lists/admin/subscribelib2.php]: True. I guess you can check if it contains only numbers and (optional) commas, like /(\d+,*)+/

**Comment by samtuke** [comment]
@xh3n1 Review still in progress?

**Comment by xh3n1** [comment]
@samtuke Waiting for @lwcorp to add the check, but it's still ok to be merged.

**Comment by samtuke** [comment]
@lwcorp Are you planning to make the discussed amendment?

**Comment by lwcorp** [comment]
Not sure I understand. @xh3n1 asked me a yes/no question 8 days ago and I replied to it. @xh3n1 can you follow my answer and tell us how to proceed?

File: public_html/lists/admin/subscribelib2.php (hunk 1 above)
Comment (xh3n1) [reply public_html/lists/admin/subscribelib2.php]: @lwcorp yes-  that's what I had in mind. Would it be ok with you implementing the change?

Comment (lwcorp) [reply public_html/lists/admin/subscribelib2.php]: How do I edit my commit? All you need to do is change:
`} elseif (!isset($_POST["subscribe"]) && isset($_GET['list'])) {`
into:
`} elseif (!isset($_POST["subscribe"]) && isset($_GET['list']) && preg_match("/^(\d+,)*\d+$/", $_GET['list'])) {`

**Comment by samtuke** [comment]
Just push additional commits to the same branch in order to update this pull request. The new commits will show up on this page automatically. Eg commit and push in the same way you did for the original pull request.

**Comment by lwcorp** [comment]
The problem is it's "unknown":
image

Does it mean I have to submit the patch from scratch?

**Comment by samtuke** [comment]
That's very strange - no, I'll merge this, then you can create a separate PR without having to re-do the existing changes. Thanks!

**Comment by lwcorp** [comment]
It's because I did the original pull directly without forking hoping it won't require edits since I've been using this code for years.
This time I forked so you can find your requested data verification at https://github.com/phpList/phplist3/pull/543

**Comment by samtuke** [comment]
Great, thank you. It will be reviewed shortly.

**Pull request merged**